ada_feeding_perception/ada_feeding_perception/segment_from_point.py

```diff
@@ -260,11 +307,22 @@ def segment_image(
         # Create the result
         result = SegmentFromPoint.Result()
         result.header = image_msg.header
+        with self.camera_info_lock:
+            result.camera_info = self.camera_info
+
+        # Get the latest depth image
+        with self.latest_depth_img_msg_lock:
+            depth_img_msg = self.latest_depth_img_msg
 
         # Convert the image to OpenCV format
         image = self.bridge.imgmsg_to_cv2(image_msg, desired_encoding="bgr8")
         image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
 
+        # Convert the depth image to OpenCV format
```

Comment:
Nit: Probably best to reiterate as a comment here that the data format is 16bit Integer mm.